### PR TITLE
Improve csrf token handeling

### PR DESCRIPTION
**Pull request by alexdashkov**
# Description
TLDR: 
Do not execute an extra call. Do not change headers on every post/put/delete request.

This PR is an addition/improvement for #23 and #22. 
1. A call to the `login` endpoint does not require a CSRF token and adds it to the cookies.
2. Login endpoint override admin endpoint token so we can't use it anymore and should update it.
Following this behavior, we can just remove a call to the admin endpoint and instead use only one call to the `login` endpoint. Just after that, we put the token from cookies to headers and as the result, all the post/put/delete calls from this session will have the right token.

Changes in this PR: 
- Do not call the admin endpoint to get a token
- Do not update csrf token for every call

### CODE REVIEW

**Review by setu4993 — APPROVED** [review]
Thanks, @alexdashkov!

**Comment by setu4993** [comment]
@Hironsan : When you get a chance to review / merge this, can you please also push a tag with the associated merge that'll trigger the flow to upload the package to PyPI? I do not see an upload after #23 got merged...
